### PR TITLE
Update configuration docs

### PR DESCRIPTION
**Pull request by iddan**
- Added paragraph about default configuration
 - Added more structure
 - Organized by priority
 - Added index of contents

---
This change is [Reviewable](https://reviewable.io/reviews/cayleygraph/cayley/840)

### CODE REVIEW

File: docs/Configuration.md [hunk 1]
```diff
@@ -1,206 +1,206 @@
-# Configuration File
+# Configuration
 
-## Overview
+Cayley can be configured using configuration file written in YAML / JSON or by passing flags to the `http` command line.
```
Comment (dennwc) [inline docs/Configuration.md]: It's not specific to `http`, it works for other commands as well: `repl`, `load`, `init`, `dump`.

File: docs/Configuration.md [hunk 2]
```diff
@@ -1,206 +1,206 @@
...
 
-Cayley looks in the following locations for the configuration file (named `cayley.yml` or `cayley.json`):
+## Recommended Configuration
 
-  * Command line flag
-  * The environment variable $CAYLEY_CFG
-  * Current directory.
-  * $HOME/.cayley/
-  * /etc/
+By default, Cayley is using an on-disk `bolt` store. This will best work for most use cases.
```
Comment (dennwc) [inline docs/Configuration.md]: No, I think by default it uses `memstore`, but `bolt` is indeed recommended for a basic persistent configuration.

File: docs/Configuration.md [hunk 3]
```diff
@@ -1,206 +1,206 @@
...
-  * `pouch`: Stores the graph data and indices in a [PouchDB](https://pouchdb.com/). Requires building with [GopherJS](https://github.com/gopherjs/gopherjs).
-
-  **SQL backends**
-  
-  * `postgres`: Stores the graph data and indices in a [PostgreSQL](https://www.postgresql.org) instance.
-  * `cockroach`: Stores the graph data and indices in a [CockroachDB](https://www.cockroachlabs.com/product/cockroachdb/) cluster.
-  * `mysql`: Stores the graph data and indices in a [MySQL](https://www.mysql.com/) or [MariaDB](https://mariadb.org/) instance.
-  * `sqlite`: Stores the graph data and indices in a [SQLite](https://www.sqlite.org) database.
+- Type: String
+- Default: `"bolt"`
```
Comment (dennwc) [inline docs/Configuration.md]: Same here, wrong default.

File: docs/Configuration.md (hunk 2 above)
Comment (iddan) [reply docs/Configuration.md]: I think the Docker is inconsistent with the CLI. Should we fix the Docker, or change the CLI?

Comment (iddan) [reply docs/Configuration.md]: I opened a different PR to fix the Dockerfile: https://github.com/cayleygraph/cayley/pull/842

**Pull request merged**